metasynth/__main__.py

```diff
@@ -1,56 +1,86 @@
 """CLI for generating synthetic data frames from a metasynth .json file."""
 import argparse
+import json
 import pathlib
 import pickle
 import sys
 
-from metasynth._version import __version__
+try:  # Python < 3.10 (backport)
+    from importlib_metadata import entry_points
+except ImportError:
+    from importlib.metadata import entry_points  # type: ignore [assignment]
+
 from metasynth import MetaFrame
+from metasynth._version import version_tuple
+from metasynth.validation import create_schema
+
+SEMVER = f"{version_tuple[0]}.{version_tuple[1]}.{version_tuple[2]}"
+
+MAIN_HELP_MESSAGE = f"""
+Metasynth CLI version {SEMVER}
+
+Usage: metasynth [subcommand] [options]
+
+Available subcommands:
+    synthesize - generate synthetic data from a .json file
+    jsonschema - generate json schema from distribution providers
+
+Program information:
+    -v, --version - display CLI version and exit
+    -h, --help    - display this help file and exit
+"""
+
+ENTRYPOINTS = ["synthesize", "jsonschema"]
+
 
+def main() -> None:
+    """CLI pointing to different entrypoints."""
+    # show help by default, else consume first argument
+    subcommand = "--help" if len(sys.argv) < 2 else sys.argv.pop(1)
```

Comment:
It looks like argparse has functionality for sub-commands natively. Or is there a reason to do it manually?

https://docs.python.org/3/library/argparse.html#sub-commands

Reply:
won't fix in this PR if that's ok. Could not get the help file to display how I want. We can fix it if we want to add a new entrypoint or this file becomes more complex.